pkg/migration-tool/reader/reader.go

```diff
@@ -93,5 +117,133 @@ func (rr *RemoteRead) Run(errChan chan<- error) {
 			rr.sigBlockRead <- blockRef
 			blockRef = nil
 		}
-	}()
+	}
+
+	readPromBlocks := func() {
+		// Comment.
+		defer deferFunc()
+		db, err := tsdb.OpenDBReadOnly(rr.path, nil)
+		if err != nil {
+			errChan <- fmt.Errorf("remote-read run: opening prom tsdb: %w", err)
+			return
+		}
+		defer func() {
+			err = tsdb_errors.NewMulti(err, db.Close()).Err()
+		}()
+		querier, err := db.Querier(rr.c, rr.plan.Mint(), rr.plan.Maxt())
+		if err != nil {
+			errChan <- fmt.Errorf("remote-read run: prom tsdb querier: %w", err)
+			return
+		}
+		defer querier.Close()
+		ss := querier.Select(false, nil, labels.MustNewMatcher(labels.MatchRegexp, "", ".*"))
+		var iteratorStore = new(sync.Map)
+		type cusMap struct {
+			labels labels.Labels
+			itr    chunkenc.Iterator
+		}
+		// Fill the iteratorStore with the iterators across all the series-sets for the given time-range.
+		for ss.Next() {
+			series := ss.At()
+			iteratorStore.Store(series.Labels().String(), cusMap{labels: series.Labels(), itr: series.Iterator()})
+		}
+		// Start creating blocks and filling them with samples.
+		for rr.plan.ShouldProceed() {
+			select {
+			case <-rr.c.Done():
+				return
+			case <-rr.SigForceStop:
+				return
+			default:
+			}
+			blockRef, err := rr.plan.NextBlock()
+			if err != nil {
+				errChan <- fmt.Errorf("remote-run run: %w", err)
+				return
+			}
+			ts := make(tsMap)
+			mint, maxt := blockRef.Mint(), blockRef.Maxt()
+			callBack := func(_ interface{}, value interface{}) bool {
+				cus, ok := value.(cusMap)
+				if !ok {
+					errChan <- fmt.Errorf("remote-read run: value not found in iteratorStore")
+					return false
+				}
+				itr := cus.itr
+				var samples []prompb.Sample
+				appender := func() bool {
+					t, v := itr.At()
+					if t >= mint && t < maxt {
+						samples = append(samples, prompb.Sample{
+							Timestamp: t,
+							Value:     v,
+						})
+						return true
+					}
+					return false
+				}
+				if itr.Seek(mint) {
+					appender()
+					for itr.Next() {
+						if !appender() {
+							break
+						}
+					}
+				}
+				if len(samples) > 0 {
+					ts[cus.labels.String()] = struct {
+						labels  []prompb.Label
+						samples []prompb.Sample
+					}{labels: labelsToPrombLabels(cus.labels), samples: samples}
+				}
+				return true
+			}
+			iteratorStore.Range(callBack)
+			timeseries, bytesUncompressed := tsMapToTs(ts)
+			ts = nil
+			blockRef.Fill(timeseries, bytesUncompressed)
+			//fmt.Println("Sending")
+			rr.sigBlockRead <- blockRef
+			blockRef = nil
+		}
+	}
+	switch rr.readType {
```

Comment:
This seems really strange. I would have expected usage of different readers instead. e.g. RemoteReader->RemoteReaderURL, PathReader